mgmtsystem/azure.py

```diff
@@ -96,21 +98,42 @@ def stop_vm(self, vm_name, resource_group=None, shutdown=False):
             self.run_script(
                 "Stop-AzureRmVm -ResourceGroup \"{}\" -Name \"{}\" -Force"
                 .format(resource_group or self.resource_group, vm_name).strip())
+            self.wait_vm_stopped(vm_name, resource_group)
         else:
             self.logger.info("Azure VM {} is already stopped".format(vm_name))
 
+    def suspend_vm(self, vm_name, resource_group=None):
+        if self.is_vm_running(vm_name, resource_group or self.resource_group):
+            self.logger.info("Attempting to Suspend Azure VM {}".format(vm_name))
+            self.run_script(
+                "Stop-AzureRmVm -ResourceGroup \"{}\" -Name \"{}\" -Force -StayProvisioned"
+                .format(resource_group or self.resource_group, vm_name).strip())
+            self.wait_vm_suspended(vm_name, resource_group)
+        else:
+            self.logger.info("Azure VM {} is already suspended or stopped".format(vm_name))
+
     def wait_vm_running(self, vm_name, resource_group=None, num_sec=300):
         wait_for(
             lambda: self.is_vm_running(vm_name, resource_group or self.resource_group),
             message="Waiting for Azure VM {} to be running.".format(vm_name),
             num_sec=num_sec)
 
+    def wait_vm_steady(self, vm_name, resource_group=None, num_sec=300):
+        self.logger.info("All states are steady in Azure. {}".format(vm_name))
```

Comment:
It's because vm.py erroneous thinks it can't perform certain operations on Azure until azure is in some "steady" state.  I've cleaned up vm.py to the extent I can to ignore these if provider.type = azure.  This method was never even included in azure.py, so it's hard to say I'm changing anything other than clean-up exceptions and poor performance.  Bottom line is this should never be called, but unfortunately other providers seem to need it.  All the correct waits (starting, stopping, running, etc.) should be used explicitly, and this method will catch and ignore any other calls from the integration tests.

See cfme/common/vm.py delete_from_provider for what I'm trying to avoid.

    def delete_from_provider(self):
        if self.provider.mgmt.does_vm_exist(self.name):
            try:
                if self.provider.mgmt.is_vm_suspended(self.name) and self.provider.type != 'azure':
                    # Azure doesn't require VM to be in any particular state to delete it.
                    logger.debug("Powering up VM %s to shut it down correctly on %s.",
                                self.name, self.provider.key)
                    self.provider.mgmt.start_vm(self.name)
                    self.provider.mgmt.wait_vm_steady(self.name)
                    self.provider.mgmt.stop_vm(self.name)
                    self.provider.mgmt.wait_vm_steady(self.name)

Anyway, I'm happy to delete it, or put it back to not implemented.